src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs

```diff
@@ -718,4 +794,378 @@ protected override void OnCollectionChanged(NotifyCollectionChangedEventArgs e)
 		}
 
 	}
+#if __IOS__
+	#region "Helper classes for the Modal Pop Up"
+	public partial class MultiFrame : Grid
+	{
+		private readonly TaskCompletionSource<bool> _isReady = new TaskCompletionSource<bool>();
+
+#if WINUI
+		private DispatcherQueue _dispatcher => DispatcherQueue;
+#else
+		private CoreDispatcher _dispatcher => Dispatcher;
+#endif
+
+		public MultiFrame()
+		{
+			Loaded += OnLoaded;
+		}
+
+		private void OnLoaded(object sender, RoutedEventArgs e)
+		{
+			_isReady.TrySetResult(true);
+		}
+
+		public async Task OpenModal(FrameSectionsTransitionInfo transitionInfo, Page page) // Runs on background thread.
+		{
+#if __IOS__
+
+			var uiViewController = new UiViewController(page);
+
+			var rootController = UIApplication.SharedApplication.KeyWindow.RootViewController;
+
+			await rootController.PresentViewControllerAsync(uiViewController, animated: false);
+#else
+			await Task.CompletedTask;
+#endif
+		}
+
+		public async Task CloseModal()
+		{
+#if __IOS__
+			try
+			{
+				var rootController = UIApplication.SharedApplication.KeyWindow.RootViewController;
+
+				await rootController.DismissViewControllerAsync(false);
+			}
+			catch (Exception) { /* purposely */ }
+#else
+			await Task.CompletedTask;
+#endif
+		}
+
+		public class UiViewController : _UIViewController
+		{
+			public UiViewController(Page frame)
+			{
+#if __IOS__
+				View = frame;
+#endif
+			}
+
+			public UIViewControllerSectionsTransitionInfo OpeningTransitionInfo { get; set; }
+
+			public void SetTransitionInfo(UIViewControllerSectionsTransitionInfo transitionInfo)
+			{
+				ModalInPresentation = !transitionInfo.AllowDismissFromGesture;
+				ModalPresentationStyle = transitionInfo.ModalPresentationStyle;
+				ModalTransitionStyle = transitionInfo.ModalTransitionStyle;
+			}
+		}
+
+		public abstract class FrameSectionsTransitionInfo : SectionsTransitionInfo
```

Comment:
you can probably trim down the fat a little, some of these stock transitions arent used at all, even more so if you removed no-ios code

Reply:
did that thanks